contest/views.py

```diff
@@ -76,63 +76,64 @@ def new(request):
                 new_contest = form.save()
                 logger.info('Contest: Create a new contest %s!' % new_contest.id)
                 return HttpResponseRedirect('/contest/')
-    else:
-        raise PermissionDenied
+    raise PermissionDenied
     
 
 def edit(request,contest_id):
-    try:
-        contest = Contest.objects.get(id = contest_id)
-    except Contest.DoesNotExist:
-        logger.warning('Contest: Can not edit contest %s! Contest not found!' % contest_id)
-        raise Http404('Contest does not exist, can not edit.')
+    if request.user.is_authenticated():
+        try:
+            contest = Contest.objects.get(id = contest_id)
+        except Contest.DoesNotExist:
+            logger.warning('Contest: Can not edit contest %s! Contest not found!' % contest_id)
+            raise Http404('Contest does not exist, can not edit.')
 
-    if user_info.has_c_ownership(request.user,contest):
-        if request.method == 'GET':        
-            contest_dic = model_to_dict(contest)
-            form = ContestForm(initial = contest_dic)
-            return render(request,'contest/editContest.html',{'form':form,'user':request.user})
-        if request.method == 'POST':
-            form = ContestForm(request.POST, instance = contest)
-            if form.is_valid():
-                modified_contest = form.save()
-                logger.info('Contest: Modified contest %s!' % modified_contest.id)
-            return HttpResponseRedirect('/contest/')
-    else:
-        raise PermissionDenied
+        if user_info.has_c_ownership(request.user,contest):
```

Comment:
May I ask that the meaning of has_c_ownership?
I guess that is to check whether an user has ownership of a contest.
If so, I prefer to name it as has_contest_ownership.
It will be more clear =)